libyul/ObjectParser.cpp

```diff
@@ -104,10 +127,92 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping() const
+{
+	return tryGetSourceLocationMapping(m_scanner->currentCommentLiteral());
+}
+
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping(std::string const& _text)
+{
+	// @use-src 0:"abc.sol" , "1:foo.sol" ,2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@use-src\s+(.*)$)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	string_view text(_text);
+	std::cmatch cm;
+	if (!std::regex_search(text.data(), text.data() + text.size(), cm, lineRE))
+		return nullopt;
+	solAssert(cm.size() == 3, "");
+
+	// Let @c text point to the parameter value (last match).
+	text = string_view(cm[2].first, static_cast<size_t>(std::distance(cm[2].first, cm[2].second)));
+
+	// iteratively match for NUM : STRING_LITERAL and increment
+	static regex const firstParamRE(R"~~~(\s*(\d+)\s*:\s*"((?:\\\"|[^\"])*)")~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	static regex const continuationParamRE(R"~~~(\s*,\s*(\d+)\s*:\s*"((?:\\\"|[^\"])*)")~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+
+	ReverseSourceNameMap result;
+
+	int k = 0;
```

Comment:
Is `k` the index?

Reply:
yes, it is used to determine whether or not to check for a leading comma in the list of parameters. a boolean would have done it too, I just realize.

Reply:
k-index completely removed.